config/v1/types_infrastructure.go

```diff
@@ -631,23 +631,18 @@ type GCPPlatformStatus struct {
 	// +openshift:enable:FeatureSets=CustomNoUpgrade;TechPreviewNoUpgrade
 	ResourceTags []GCPResourceTag `json:"resourceTags,omitempty"`
 
-	// clusterHostedDNS indicates the type of DNS solution in use within the cluster. Its default value of
-	// "Disabled" indicates that the cluster's DNS is the default provided by the cloud platform. It can be
-	// "Enabled" during install to bypass the configuration of the cloud default DNS. When "Enabled", the
-	// cluster needs to provide a self-hosted DNS solution for the cluster's installation to succeed.
-	// The cluster's use of the cloud's Load Balancers is unaffected by this setting.
-	// The value is immutable after it has been set at install time.
-	// Currently, there is no way for the customer to add additional DNS entries into the cluster hosted DNS.
-	// Enabling this functionality allows the user to start their own DNS solution outside the cluster after
-	// installation is complete. The customer would be responsible for configuring this custom DNS solution,
-	// and it can be run in addition to the in-cluster DNS solution.
-	// +kubebuilder:default:="Disabled"
-	// +default="Disabled"
-	// +kubebuilder:validation:Enum="Enabled";"Disabled"
-	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="clusterHostedDNS is immutable and may only be configured during installation"
-	// +optional
+	// cloudLoadBalancerConfig contains the IP addresses of API, API-Int and
+	// Ingress Load Balancers created on the cloud platform. Its values would
+	// not be populated for on-prem platforms. These Load Balancer IPs are
+	// used to configure the in-cluster DNS instance for API, API-Int and
+	// Ingress services. `clusterHostedDNS` is expected to be `Enabled` when
+	// these Load Balancer IP addresses are populated and used.
+	//
+	// +default={"dnsType": "PlatformDefault"}
+	// +kubebuilder:default={"dnsType": "PlatformDefault"}
 	// +openshift:enable:FeatureSets=CustomNoUpgrade;TechPreviewNoUpgrade
-	ClusterHostedDNS ClusterHostedDNS `json:"clusterHostedDNS,omitempty"`
+	// +optional
+	CloudLoadBalancerConfig *CloudLoadBalancerConfig `json:"cloudLoadBalancerConfig,omitempty"`
```

Comment:
I don't think this needs to be a pointer, it doesn't contain any required fields. What happens if you try and create an infra with `cloudLoadBalancerConfig: {}`, I assume the defaulting kicks in and everything is happy?

Reply:
Further discussion, to allow successful defaulting, this needs to be a pointer with omitempty